### DISCUSSION TITLE
Support for STM32f412ZGTx -Problem

### DISCUSSION THREAD

**Discussion by muellevin** [stm32duino/Arduino_Core_STM32#1962] [Q&A] [answered]
Hii am trying to add the generic Variant for the STM32F412ZGTx (Nucleo-144) Board. In the [wiki](https://github.com/stm32duino/Arduino_Core_STM32/wiki/Add-a-new-variant-%28board%29#3---generic-system-clock-configuration) the HSI Clock is configured to be used. But i get an error in STM32CubeMX (Version 6.7.0) as it says the source for the PLL Mux should be the HSE Clock instead. Would there be any Problem resulting from changing the Clock from HSI to HSE?

**Comment by fpistm** [comment, marked as answer]
It is only a warning. If your board doesn't have an HSE it will not work.

**Reply by muellevin** [reply]
In the wiki is written `For generic board the internal clock is used: HSI` and further `Set PLL Source Mux to HSI` this produces the warning. I can change the `PLL Source Mux to HSE` and the warning disappears. The question is can i simply ignore the Warning or should i use the HSE Clock.

My target is to use the STM32 in the Arduino IDE, since a couple of needed libs are only written there.

**Reply by muellevin** [reply]
Hello are there any new information about it?

**Reply by fpistm** [reply]
Generic variants are designed to run on the targeted mcu that's why internal clock source is set to HSI.
So if you want provide the generic support of the STM32F412ZGTx you have to use HSI.
As it seems you have a nucleo F412ZG:
https://www.st.com/en/evaluation-tools/nucleo-f412zg.html
This boards doesn't have an HSE but can use HSE bypass as the STLink can provide it. In this case you can create a dedicated variant for this board with dedicated system core clock config.
![image](https://user-images.githubusercontent.com/20641798/223370787-a48828f0-b1b0-4783-84c6-65f607476234.png)



About the warning as stated it is only a warning.

Here example of generic F429ZI and Nucleo F429ZI support:
https://github.com/stm32duino/Arduino_Core_STM32/tree/main/variants/STM32F4xx/F427Z(G-I)T_F429ZET_F429Z(G-I)(T-Y)_F437Z(G-I)T_F439Z(G-I)(T-Y)
